Player JFrame added

MelissaRoman06 opened the pull request:
Please give your comments on Player class.

MelissaRoman06 committed:
Player JFrame added

MelissaRoman06 committed:
PlayerMain and PlayerJFrame classes split

MelissaRoman06 committed:
Panel's names changed to more descriptive ones

MelissaRoman06 committed:
Boxes' names changed to more descriptive ones

MelissaRoman06 committed:
Copyright added

aburgosu commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+    private boolean AccionProgres = true;
+
+    private EmbeddedMediaPlayerComponent player = new EmbeddedMediaPlayerComponent();
+    private File file1 = new File("resources/Wildlife.wmv");
```
edit variable name file1

MelissaRoman06 replied:
Variable name changed

aburgosu commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+        new ButtonListener();
+    }
+
+    class ButtonListener {
```
Divide the classes. Only one class for file.

MelissaRoman06 replied:
New separate ButtonListener class created

aburgosu commented on src/main/java/com/fundation/logic/view/PlayerMain.java:
```diff
@@ -0,0 +1,24 @@
+public class PlayerMain {
+
+    public static void main(String[] args) {
+
```
delete empty line

MelissaRoman06 replied:
Empty line according on our code conventions

titor-oopart commented on src/main/java/com/fundation/logic/view/Player.java:
```diff
@@ -0,0 +1,194 @@
+        new ButtonListener();
+    }
+
+    class ButtonListener {
```
create other file for this class

MelissaRoman06 replied:
outdated

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -1,11 +1,18 @@
 import uk.co.caprica.vlcj.player.MediaPlayer;
 import uk.co.caprica.vlcj.player.MediaPlayerEventAdapter;
 import uk.co.caprica.vlcj.runtime.RuntimeUtil;
-
```
Delete this line

MelissaRoman06 replied:
Done

titor-oopart commented on src/main/java/com/fundation/logic/view/Player.java:
```diff
@@ -0,0 +1,194 @@
+    }
+}
+
+class PlayerJFrame extends JFrame {
```
create other file for each class that is not the main class

MelissaRoman06 replied:
outdated

titor-oopart commented on src/main/java/com/fundation/logic/view/Player.java:
```diff
@@ -0,0 +1,194 @@
+        public ButtonListener() {
+
+            playButton.addMouseListener(new MouseAdapter() {
+                public void mousePressed(MouseEvent e) {
```
Please comment all methods for know what are doing

MelissaRoman06 replied:
Done

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerMain.java:
```diff
@@ -1,5 +1,19 @@
+ * @version 1.0
+ */
 public class PlayerMain {
 
```
Delete the spaces in blanks.

MelissaRoman06 replied:
Blanks according on author tag example

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -14,6 +21,12 @@
+ * @version 1.0
+ */
 class PlayerJFrame extends JFrame {
 
```
Delete this spaces in blanks.

MelissaRoman06 replied:
Blanks according on author tag example

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -44,7 +57,6 @@ public PlayerJFrame() {
 
         setTitle("LogicalPlayer");
 
```
Delete Spaces in blank.

MelissaRoman06 replied:
Done

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -1,11 +1,18 @@
 import uk.co.caprica.vlcj.player.MediaPlayerEventAdapter;
 import uk.co.caprica.vlcj.runtime.RuntimeUtil;
-
 import javax.swing.*;
```
Don't use *

MelissaRoman06 replied:
Done

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+                    }
+                }
+            });
+
```
Delete this line

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+                    AccionProgres = true;
+                }
+            });
+
```
Delete this line.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+                public void mousePressed(MouseEvent e) {
+                    AccionProgres = false;
+                }
+
```
Delete this line.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+                    player.getMediaPlayer().setPause(player.getMediaPlayer().isPlaying() ? true : false);
+                }
+            });
+
```
Delete this line.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+                    progressBar.setEnabled(false);
+                }
+            });
+
```
Delete this line.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+                    System.out.println(vol);
+                }
+            });
+
```
Delete this line.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+                    setTitle(file1.getName() + " - LogicalPlayer");
+                }
+            });
+
```
Delete this line.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+    class ButtonListener {
+
+        public ButtonListener() {
+
```
Delete this line.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+        } catch (NullPointerException e) {
+            System.out.println("Icons cannot be found..");
+        }
+
```
Delete this line.

MelissaRoman06 replied:
Done

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+    private File file1 = new File("resources/Wildlife.wmv");
+
+    static {
+        NativeLibrary.addSearchPath(RuntimeUtil.getLibVlcLibraryName(), "C:/Program Files/VideoLAN/VLC/");
```
Don't use a generic path.

MelissaRoman06 replied:
It is necessary to use a relative path because the fileToBePlayed may not exist in the absolute path on other person system

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+    private JLabel stopButton = new JLabel();
+    private JLabel pauseButton = new JLabel();
+    private int vol = 50;
+
```
Delete space in Blanks.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+    private JPanel headerPanel = new JPanel();
+    private JPanel playingPanel = new JPanel();
+    private JPanel bottomPanel = new JPanel();
+
```
Delete space in Blanks.

MelissaRoman06 replied:
Empty line in order to separate JPanel declarations from other

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+
+    private JSlider progressBar = new JSlider(0, 100, 0);
+    private JSlider volumen = new JSlider(0, 100, vol);
+
```
Delete space in Blanks.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+    private JSlider volumen = new JSlider(0, 100, vol);
+
+    private boolean AccionProgres = true;
+
```
Delete space in Blanks.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+    static {
+        NativeLibrary.addSearchPath(RuntimeUtil.getLibVlcLibraryName(), "C:/Program Files/VideoLAN/VLC/");
+    }
+
```
Delete space in Blanks.

MelissaRoman06 replied:
Empty lines according on our code conventions

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+    }
+
+    public PlayerJFrame() {
+
```
Delete space in Blanks.

MelissaRoman06 replied:
Empty line according on our code conventions

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+
+        setBounds(new Rectangle(80, 200, 800, 600));
+        setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+
```
Delete space in Blanks.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+        Box pauseButtonBox = Box.createVerticalBox();
+        Box stopButtonBox = Box.createVerticalBox();
+        Box progressBarBox = Box.createVerticalBox();
+
```
Delete space in Blanks.

MelissaRoman06 replied:
Empty line in order to separate box declaration related code from other

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+        headerPanel.setBackground(Color.BLUE);
+        headerBox.setPreferredSize(new Dimension(100, 50));
+        headerPanel.add(headerBox);
+
```
Delete space in Blanks.

MelissaRoman06 replied:
Empty line in order to separate one panel related code from another

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+
+        add(bottomPanel, BorderLayout.SOUTH);
+        bottomPanel.setBackground(Color.BLUE);
+
```
Delete space in Blanks.

MelissaRoman06 replied:
Done

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+        playButtonBox.setPreferredSize(new Dimension(50, 50));
+        playButtonBox.add(playButton);
+        bottomPanel.add(playButtonBox);
+
```
Delete space in Blanks.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+        pauseButtonBox.setPreferredSize(new Dimension(50, 50));
+        pauseButtonBox.add(pauseButton);
+        bottomPanel.add(pauseButtonBox);
+
```
Delete space in Blanks.

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+        stopButtonBox.setPreferredSize(new Dimension(50, 50));
+        stopButtonBox.add(stopButton);
+        bottomPanel.add(stopButtonBox);
+
```
Delete space in Blanks.

MelissaRoman06 replied:
Empty lines in order to separate one button related code from another

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+        progressBar.setEnabled(false);
+        progressBarBox.add(volumen);
+        bottomPanel.add(progressBarBox);
+
```
Delete space in Blanks.

MelissaRoman06 replied:
Empty lines in order to separate one button related code from another

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+
+        add(playingPanel, BorderLayout.CENTER);
+        playingPanel.setBackground(Color.BLACK);
+
```
Delete space in Blanks.

MelissaRoman06 replied:
Empty lines in order to separate one panel related code from another

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+        add(player);
+        player.setSize(playingPanel.getSize());
+        player.setVisible(true);
+
```
Delete space in Blanks.

MelissaRoman06 replied:
Empty lines in order to separate one panel related code from another

TheYisusGithub commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,178 @@
+        player.setVisible(true);
+
+        setVisible(true);
+
```
Delete space in Blanks.

MelissaRoman06 replied:
Empty lines in order to separate one panel related code from another

MelissaRoman06 committed:
ButtonListener class created. Clean code improvements

MelissaRoman06 committed:
Corrections according on code review requests

MelissaRoman06 committed:
Empty lines deleted according on code review requests

MelissaRoman06 committed:
PlayerMain class (used for testing purposes only) deleted

titor-oopart commented on src/main/java/com/fundation/logic/view/ButtonListener.java:
```diff
@@ -0,0 +1,100 @@
+        this.vol = vol;
+        actionProgress = true;
+    }
+
```
Please comment what do the methods

MelissaRoman06 replied:
Done

titor-oopart commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,107 @@
+    }
+}
+
+
```
Delete this empty line

MelissaRoman06 replied:
Done

titor-oopart commented on src/main/java/com/fundation/logic/view/PlayerJFrame.java:
```diff
@@ -0,0 +1,107 @@
+    private int vol = 50;
+    private JPanel headerPanel = new JPanel();
+    private JPanel playingPanel = new JPanel();
+    private JPanel bottomPanel = new JPanel();
```
Please separate all declared object of the initialize objects

MelissaRoman06 replied:
Done

MelissaRoman06 committed:
Corrections according on code review requests

In PlayerJFrame class, declaration and initilization separated as required
Comments added to explain what methods do

MelissaRoman06 committed:
VLC Portable added to thirdparty package. But the code does not recognize corectly the libraries. VLC installation needed.

MelissaRoman06 committed:
Misspelling corrections on method's comment
